### ISSUE TITLE
Normalization of the data

### ISSUE THREAD

**Issue by rodrigoberriel** [EliSchwartz/DeltaEncoder#4]
This is not an actual issue; it is just a question.

In the Section 3.1 of the paper, it is stated that:

> For both models the head, i.e., the layers after the last convolution, is replaced by two fully-connected layers with 2048 units with ReLU activations. The features used are the 2048-dimensional outputs of the last fully connected layer.

Based on that, I was expecting the data to be in the range `[0, inf]` (after ReLU). Then, I ran:

```python
import pickle
features_train, labels_train, features_test, labels_test, episodes_1shot, episodes_5shot = pickle.load(open('data/mIN.pkl','rb'))

print("features_train range: [{}, {}]".format(features_train.min(), features_train.max()))
print("features_test range: [{}, {}]".format(features_test.min(), features_test.max()))
print("episodes_1shot range: [{}, {}]".format(min([x[0].min() for x in episodes_1shot]), max([x[0].max() for x in episodes_1shot])))
print("episodes_5shot range: [{}, {}]".format(min([x[0].min() for x in episodes_5shot]), max([x[0].max() for x in episodes_5shot])))
```

and the output was:

> features_train range: [-1.0, 1.0]
> features_test range: [-1.0, 0.534611165523529]
> episodes_1shot range: [-1.0, 0.12915128469467163]
> episodes_5shot range: [-1.0, 0.22022214531898499]

Given these ranges (particularly the `features_train` one), I assume a normalization process was applied and, in my experiments, this normalization seems to be important for the convergence of the encoder (kind of expected, fair enough). As I could not find any information about how the normalization was performed, nor in the paper, neither in this repository, would you mind to share some info on this (e.g., motivation and how it was done)?

Nice paper, BTW. Very clear and easy to read :+1:

**Comment by EliSchwartz** [comment]
Thank you for your kind words.

It is simply normalized to be in [-1,1], so:

def normalize(x, max_val, min_val)
  x -= min_val
  x /= (max_val - min_val)
  return 2*x-1

where max_val and min_val are set according to the training set:
max_val = features_train.max()
min_val = features_train.min()
